.github/actions/cache/action.yml

```diff
@@ -11,9 +11,14 @@ description: 'Caches cargo dependencies'
 runs:
   using: composite
   steps:
-    - uses: actions/cache@v3 
-      with: 
-        path: | 
-          ~/.cargo/ 
+    - uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cargo/.crates.toml
+          ~/.cargo/.crates2.json
+          ~/.cargo/bin/
+          ~/.cargo/git/db/
+          ~/.cargo/registry/cache/
+          ~/.cargo/registry/index/
```

Comment:
Can you add a comment here that explains why we do this instead of just doing `~/.cargo/` (and mentions #1312)? I know I'd personally be confused if I saw this with no explanation.

Reply:
Also, including the link you mentioned above would be a great breadcrumb: https://doc.rust-lang.org/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci